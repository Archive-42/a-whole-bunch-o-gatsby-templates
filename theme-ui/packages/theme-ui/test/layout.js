import React from 'react'
import renderer from 'react-test-renderer'
import {
  Box,
  Flex,
  Layout,
  Main,
  Container,
  Header,
  Footer,
} from '../src/layout'

test.each([
  ['Box', Box],
  ['Flex', Flex],
  ['Layout', Layout],
  ['Main', Main],
  ['Container', Container],
  ['Header', Header],
  ['Footer', Footer],
])('renders %s', (a, Component) => {
  const json = renderer.create(<Component />).toJSON()
  expect(json).toMatchSnapshot()
})
