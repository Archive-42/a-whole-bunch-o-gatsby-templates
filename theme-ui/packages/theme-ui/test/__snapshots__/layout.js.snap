// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Box 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
}

<div
  className="emotion-0"
/>
`;

exports[`renders Container 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  width: 100%;
  min-width: 0;
  max-width: 1024px;
  margin-left: auto;
  margin-right: auto;
  padding: 32px;
}

<div
  className="emotion-0"
/>
`;

exports[`renders Flex 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="emotion-0"
/>
`;

exports[`renders Footer 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<footer
  className="emotion-0"
/>
`;

exports[`renders Header 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<header
  className="emotion-0"
/>
`;

exports[`renders Layout 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="emotion-0"
/>
`;

exports[`renders Main 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

<main
  className="emotion-0"
/>
`;
